### PR TITLE
Joy/ref 5806/s3sink

### PR DESCRIPTION
**Pull request by PMjoydas**
S3 sink component

### CODE REVIEW

File: components/Python/Scikit-learn/sink/s3_file_sink/component.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
+  "engineType": "Python",
+  "userStandalone": false,
+  "language": "Python",
+  "name": "s3_file_sink",
+  "label": "S3 File Sink",
+  "description": "Save a file to an S3 bucket + key",
+  "program": "main.py",
+  "componentClass": "S3FileSink",
```
Comment (sriram109) [inline components/Python/Scikit-learn/sink/s3_file_sink/component.json]: is this field relevant?

File: components/Python/Scikit-learn/sink/s3_file_sink/main.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    def _materialize(self, parent_data_objs, user_data):
+        file_path = str(parent_data_objs[0])
+        self._save_file(file_path)
+        return []
+
+    def _save_file(self, file_path):
+
+        # Initialize mlops
+        mlops.init()
```
Comment (sriram109) [inline components/Python/Scikit-learn/sink/s3_file_sink/main.py:27]: do each component have to init mlops?

File: components/Python/Scikit-learn/sink/s3_file_sink/main.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            aws_access_key_id=self._params["aws_access_key_id"],
+            aws_secret_access_key=self._params["aws_secret_access_key"],
+        )
+
+        data = open(file_path, 'rb')
+        save_start_time = time.time()
+        client.put_object(Bucket=self._params["bucket"], Key=self._params["key"], Body=data)
+        save_elapsed_time = time.time() - save_start_time
+        if self._params["get_save_time"]:
+            mlops.set_stat("s3.outputSaveTimemsec", save_elapsed_time)
```
Comment (sriram109) [inline components/Python/Scikit-learn/sink/s3_file_sink/main.py:48]: is latency useful or do we need to look at the the throughput? (just a thought :))

**Review by sriram109 — APPROVED** [review]

File: components/Python/Scikit-learn/sink/s3_file_sink/main.py (hunk 2 above)
Comment (PMjoydas) [reply components/Python/Scikit-learn/sink/s3_file_sink/main.py:27]: At least one component in the pipeline needs to perform the "mlops.init()" 
given these are connectable components not clear what other components are (or aren't) performing the init.

however calling into mlops.init() by each of the component isn't an issue, from the usage prospective, not clear on the resource concerns if any on the server.. (experts can comment)

File: components/Python/Scikit-learn/sink/s3_file_sink/main.py (hunk 3 above)
Comment (PMjoydas) [reply components/Python/Scikit-learn/sink/s3_file_sink/main.py:48]: can always provide the inverse of latency

**Review by yakov-g — APPROVED** [review]

**Pull request merged**